lib/inspector.js

```diff
@@ -128,13 +129,26 @@ class Session extends EventEmitter {
     }
     const id = this[nextIdSymbol]++;
     const message = { id, method };
+    const configureCbAndDispatch = (cb) => {
+      this[messageCallbacksSymbol].set(id, cb);
+      this[connectionSymbol].dispatch(JSONStringify(message));
+    };
+
     if (params) {
       message.params = params;
     }
     if (callback) {
-      this[messageCallbacksSymbol].set(id, callback);
+      configureCbAndDispatch(callback);
+      return;
     }
-    this[connectionSymbol].dispatch(JSONStringify(message));
+
+    const event = new EventEmitter();
+    const kOnFinish = Symbol('kOnFinish');
+    const promisifiedResult = promisify(event.once.bind(event))(kOnFinish);
+    const cb = (err, res) => event.emit(kOnFinish, err, res);
+    configureCbAndDispatch(cb);
+
+    return promisifiedResult;
```

Comment:
i don't think we need to create all this event and promisify stuff in the middle? 
```suggestion
    return new Promise((resolve, reject) => {
      configureCbAndDispatch((err, res) => {
        if (err) reject(err);
        else resolve(res);
      });
    });
```

Reply:
I loved it! I try doing with much more lines as I thought I’d be better for maintenance but for performance reasons I’d go with what you’ve suggested 🤩

already committed it!